src/number_theory/ramification_inertia.lean

```diff
@@ -0,0 +1,195 @@
+/-
+Copyright (c) 2021 Anne Baanen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen
+-/
+
+import algebra.is_prime_pow
+import field_theory.separable
+import linear_algebra.free_module.finite.rank
+import linear_algebra.free_module.pid
+import linear_algebra.matrix.nonsingular_inverse
+import ring_theory.dedekind_domain.ideal
+import ring_theory.localization.module
+
+/-!
+# Ramification index and inertia degree
+
+If `P : ideal S` lies over `p : ideal R` for the ring extension `f : R →+* S`
+(assuming `P` and `p` are prime or maximal where needed).
+The **ramification index** `ramification_idx f p P` is the multiplicity of `P` in `map f p`,
+and the **inertia degree** `inertia_deg f p P` is the degree of the field extension
+`(S / P) : (R / p)`.
+
+## TODO (#12287)
+
+The main theorem `sum_ramification_inertia` states that for all coprime `P` lying over `p`,
+`Σ P, ramification_idx f p P * inertia_deg f p P` equals the degree of the field extension
+`Frac(S) : Frac(R)`.
+
+## Implementation notes
+
+Often the above theory is set up in the case where:
+ * `R` is the ring of integers of a number field `K`,
+ * `L` is a finite separable extension of `K`,
+ * `S` is the integral closure of `R` in `L`,
+ * `p` and `P` are maximal ideals,
+ * `P` is an ideal lying over `p`
+We will try to relax the above hypotheses as much as possible.
+
+-/
+
+open_locale big_operators
+open_locale non_zero_divisors
+
+universes u v
+
+variables {R : Type u} [comm_ring R]
+variables {S : Type v} [comm_ring S] (f : R →+* S)
+variables (p : ideal R) (P : ideal S)
+
+open finite_dimensional
+open ideal
+open unique_factorization_monoid
+
+section dec_eq
+
+open_locale classical
```

Comment:
I agree that in principle we might avoid a conflict by removing the `open_locale` classical. I think in practice the drawbacks outweigh the benefits: Basically all theorems require a huge `decidable` instance if we drop the `classical` locale: `decidable_pred (λ (n : ℕ), ∀ (k : ℕ), map f p ≤ P ^ k → k ≤ n)`. There is also little risk of conflict with an actual decidability instance, since the predicate starts with a `∀` over an infinite set, so you'd have to actually define a specific, nontrivial decision algorithm for this specific problem.